feat: add support for derived distributions and gnome terminal themes

dmccaffery opened the pull request:
- fix(template): do not assume dash is the default shell in sh profile
- fix: use posix-compliant source (dot) and exit codes
- fix: handle null globs correctly
- feat: add support for derived platforms (like popos)
- chore: normalise return values from functions
- chore: remove unnecessary vars from  all profiles
- fix(install-minikube): ensure minikube is installed via brew
- feat: add support for themes in gnome terminal

dmccaffery committed:
fix(template): do not assume dash is the default shell in sh profile

dmccaffery committed:
fix: use posix-compliant source (dot) and exit codes

dmccaffery committed:
fix: handle null globs correctly

dmccaffery committed:
feat: add support for derived platforms (like popos)

dmccaffery committed:
chore: normalise return values from functions

dmccaffery committed:
chore: remove unnecessary vars from all profiles

dmccaffery committed:
feat: add support for themes in gnome terminal

dmccaffery commented:
derived install tested on Pop! OS: 
![image](https://user-images.githubusercontent.com/1803684/103328304-49260680-4a50-11eb-8c89-feb1f7d85cb5.png)

patrickserrano commented on src/zsh/scripts/eval/set-completion:
```diff
@@ -38,7 +38,7 @@ if [ ! -d "$AM_PROMPT/zsh/completions" ]; then
 
 # iterate over each completion
-for AM_COMPLETION in $AM_PROMPT/zsh/completions/* 2>/dev/null; do
+for AM_COMPLETION in $AM_PROMPT/zsh/completions/*; do
```
Oh nice! I noticed an issue with zsh completions today and was going to look into it when I had a minute.

dmccaffery committed:
chore: update apple silicon messaging

dmccaffery committed:
fix(starship): normalise symbols for battery indicators

dmccaffery committed:
perf: improve pstartup time, especially for zsh

patrickserrano reviewed (approved):
@dmccaffery looks like there are still some CI issues :/

I cloned and installed your branch locally and the issues I was having with zsh completions are resolved. So far I haven't run into any other issues which is a positive.

Cloning and installing this via `install.sh` did make me wonder if there's value in having a flag that would allow you to pass a GitHub url, so you could test installing via the bootstrap script? Might also help with CI since we were calling install directly there too, right?

dmccaffery commented:
Yeah; something changed in how WSL is implemented in GitHub actions; gosu no longer persists through sudo, so homebrew installer is chowning the path as root:build with u=rw,go=r -- and the clone of the homebrew repo fails.

I'll be looking into it when I have a chance.

dmccaffery committed:
build: add shellcheck for linting

dmccaffery committed:
fix(install): remove  unnecessary platform proxy scripts

dmccaffery committed:
fix(sh): remove show-virtualenv (replaced by starship)

dmccaffery committed:
refactor: remove support for minikube install (replaced by kind)

dmccaffery committed:
fix: remove bookmark support

- remove add-bookmark
- remove remove-bookmark
- remove show-bookmarks

NOTE:

The add-bookmark and remove-bookmark scripts have been broken for some
time and no one complained; rather than fix and continue to support a
feature that no one seems to be using, we are removing them.

Folks can add aliases via `edit-bashrc` and `edit-zshrc` if they want
to bookmark locations.

dmccaffery committed:
refactor(sh): remove do-repeat command

NOTE:

Now that `zsh` is supported as is the default shell for the vast
majority of users, this command is no longer necessary and was
confusing to use.

As a replacement, zsh includes a repeat command:

```zsh
repeat 5 { date }
```

For those still using bash:

```bash
for i in {1..5}; do date; done
```

For those using POSIX shells:

```sh
seq 1 5 | xargs date
```

dmccaffery committed:
style: use posix-compliant style for all scripts

- fix all lint issues detected by shellcheck
- replace script indentation spaces with tabs
  - while i hate this, it `just works` with heredocs
- remove use of locals, which are not posix-compliant
- quote all variable usage
- replace echo (which is inconsistent) with printf
- move colors to `set-colors` eval script
  - use `tput` for color coding and formatting
  - add functions and aliases for printing:
    - print-notice
    - print-success
    - print-warn
    - print-fail

dmccaffery committed:
fix(darwin): ensure the fira code directory in install

dmccaffery commented:
> @dmccaffery looks like there are still some CI issues :/
> 
> I cloned and installed your branch locally and the issues I was having with zsh completions are resolved. So far I haven't run into any other issues which is a positive.
> 
> Cloning and installing this via `install.sh` did make me wonder if there's value in having a flag that would allow you to pass a GitHub url, so you could test installing via the bootstrap script? Might also help with CI since we were calling install directly there too, right?

re: passing in a git-url -- I'm not sure I understand the ask -- do you mean to run the bootstrap script against an arbitrary url, including forks? -- if so, that is incredibly dangerous -- its a huge security risk as one could redirect the bootstrap script to do whatever it wants, even from ci/cd on a fork -- the bootstrap script also calls install directly; all the it actually does is to extract a tarball of the repo into a temp directory and then invoke the install script. The real work achieved by the bootstrap script is to download the tarball... doing that against an arbitrary repo isn't helpful, I don't think.

dmccaffery commented:
@patrickserrano : this is all that bootstrapping does:

https://github.com/automotiveMastermind/prompt/blob/d1eaccb93a7004ff382e94f2ca72b82749adb3b1/bootstrap.sh#L87-L90

The rest is just a comparison of the git-ref against the current upstream.

dmccaffery committed:
build(wsl): fix wsl build

dmccaffery commented:
@patrickserrano  / @sjk07 : should be ready for review / approval -- all builds should now pass. I had to do some gymnastics with the end-to-end build to get wsl to work again.

dmccaffery committed:
fix(darwin): fix notice block formatting for apple silicon

dmccaffery committed:
fix: formatting around heredocs

dmccaffery committed:
fix(starship): fix diverged formatting

patrickserrano commented:
> > @dmccaffery looks like there are still some CI issues :/
> > I cloned and installed your branch locally and the issues I was having with zsh completions are resolved. So far I haven't run into any other issues which is a positive.
> > Cloning and installing this via `install.sh` did make me wonder if there's value in having a flag that would allow you to pass a GitHub url, so you could test installing via the bootstrap script? Might also help with CI since we were calling install directly there too, right?
> 
> re: passing in a git-url -- I'm not sure I understand the ask -- do you mean to run the bootstrap script against an arbitrary url, including forks? -- if so, that is incredibly dangerous -- its a huge security risk as one could redirect the bootstrap script to do whatever it wants, even from ci/cd on a fork -- the bootstrap script also calls install directly; all that it actually does is to extract a tarball of the repo into a temp directory and then invoke the install script. The real work achieved by the bootstrap script is to download the tarball... doing that against an arbitrary repo isn't helpful, I don't think.

I hadn't really considered the security implications of such a change sine I was thinking of ease of manual testing larger PRs. But it makes sense why we wouldn't want to do that.

sjk07 reviewed (approved):
This is awesome stuff =]
